include/fi_util.h

```diff
@@ -643,4 +647,39 @@ int ofi_shm_map(struct util_shm *shm, const char *name, size_t size,
 		int readonly, void **mapped);
 int ofi_shm_unmap(struct util_shm *shm);
 
+/*
+ * Name Server
+ */
+
+/*
+ * Note: the signature of the util_ns_service_comp_func_t
+ * is similiar to 'rbtCompare' function pointer
+ */
+typedef int(*util_ns_service_comp_func_t)(void *svc1, void *svc2);
+typedef int(*util_ns_name_comp_func_t)(const char *name1, const char *name2);
+typedef void*(*util_ns_name_dup_func_t)(const char *dup_name);
+typedef void*(*util_ns_name_copy_func_t)(char *to_name, const char *from_name);
+typedef void*(*util_ns_name_free_func_t)(char *name);
```

Comment:
Do we expect complex name structure for providers? we may be able to take a simpler solution by adding a "name_size" field to the util_ns structure and removing the "dup", "copy" and "free" functions.